### PR TITLE
CGAL Lab: Rendering polygons with ghost edges

### PR DESCRIPTION
**Pull request by soesau**
Polygons and meshes with non-triangular faces that have ghost edges, i.e., holes connected via two identical edges to the outer border of the polygon, are now correctly rendered.

The FacetTriangulator now uses mark_domain_in_triangulation to set is_external property.

## Release Management

* Affected package(s): Polyhedron/demo

### CODE REVIEW

File: Stream_support/include/CGAL/IO/PLY/PLY_reader.h [hunk 1]
```diff
@@ -726,6 +726,9 @@ bool read_PLY_faces(std::istream& in,
 
   for(std::size_t j = 0; j < element.number_of_items(); ++ j)
   {
+    if (j == 228)
```
Comment (sloriot) [inline Stream_support/include/CGAL/IO/PLY/PLY_reader.h:729]: leftover?

File: Stream_support/include/CGAL/IO/PLY/PLY_reader.h [hunk 2]
```diff
@@ -758,6 +761,9 @@ bool read_PLY_faces(std::istream& in,
                                         PLY_property<std::vector<Integer> >(vertex_indices_tag)));
     }
 
+    if (get<0>(new_face).size() == 0)
```
Comment (sloriot) [inline Stream_support/include/CGAL/IO/PLY/PLY_reader.h:764]: same question here

File: Polyhedron/demo/Polyhedron/triangulate_primitive.h [hunk 3]
```diff
@@ -114,96 +108,142 @@ class FacetTriangulator
...
                    const Vector& normal)
   {
     P_traits cdt_traits(normal);
     cdt = new CDT(cdt_traits);
 
+    std::map<std::pair<Point, Point>, std::size_t> edge_map;
+    std::vector<bool> skip(idPoints.size(), false);
+
+    for (std::size_t i = 0; i < idPoints.size(); i++) {
+      int prev = (i - 1 + idPoints.size()) % idPoints.size();
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/triangulate_primitive.h:128]:
```suggestion
      std::size_t prev = (i - 1 + idPoints.size()) % idPoints.size();
```

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-147

**Comment by lrineau** [comment]
@sloriot You marked with https://github.com/CGAL/cgal/labels/Not%20yet%20approved. Who are the persons who should discuss this PR? That does not appear in the discussion of this PR.

**Comment by sloriot** [comment]
> @sloriot You marked with [ Not yet approved](https://github.com/CGAL/cgal/labels/Not%20yet%20approved) The feature or pull-request has not yet been approved. . Who are the persons who should discuss this PR? That does not appear in the discussion of this PR.

I wasn't sure if somebody somebody should review it

**Comment by sloriot** [comment]
I did a quick review + testing the behavior on non triangle faces and it works as before. You can merge it.

**Pull request merged**